jcl/src/java.base/share/classes/java/lang/invoke/VarHandle.java

```diff
@@ -1065,14 +1077,13 @@ DirectMethodHandleDesc getBootstrap() {
 		 * Create a nominal descriptor for a field VarHandle.
 		 * 
 		 * @param declaringClassDesc ClassDesc describing the declaring class for the field (null for array)
-		 * @param name unqualified String name of the field (null for array)
-		 * @param varHandleType ClassDesc describing the field type
+		 * @param name unqualified String name of the field (blank for array)
+		 * @param varHandleType ClassDesc describing the field type (arrayType for array)
 		 * @throws NullPointerException if there is a null argument
 		 */
 		private VarHandleDesc(Kind type, ClassDesc declaringClassDesc, String name, ClassDesc varHandleDesc) throws NullPointerException {
-			super(type.getBootstrap(), name, varHandleDesc, ConstantDescs.CD_VarHandle);
+			super(type.getBootstrap(), name, ConstantDescs.CD_VarHandle, type.getBootstrapArgs(declaringClassDesc, varHandleDesc));
```

Comment:
`NullPointerException if there is a null argument`  

Is this code missing some null checks / `Objects.requireNonNull()` calls?

Reply:
yes.. previously it was covered by the superclass. will update